# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/stat-nseconds.patch.mips
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./libc/sysdeps/linux/mips/bits/stat.h.orig	2006-08-30 09:10:29.000000000 +0200
+++ ./libc/sysdeps/linux/mips/bits/stat.h	2006-12-08 16:29:27.000000000 +0100
@@ -62,11 +62,11 @@
      * st_ctime but we don't have it under Linux.
      */
     __time_t st_atime;		/* Time of last access.  */
-    long int __reserved0;
+     unsigned long int st_atimensec;     /* Nscecs of last access.  */ 
     __time_t st_mtime;		/* Time of last modification.  */
-    long int __reserved1;
+    unsigned long int st_mtimensec;     /* Nsecs of last modification.  */
     __time_t st_ctime;		/* Time of last status change.  */
-    long int __reserved2;
+    unsigned long int st_ctimensec;     /* Nsecs of last status change.  */
     __blksize_t st_blksize;	/* Optimal block size for I/O.  */
 #ifndef __USE_FILE_OFFSET64
     __blkcnt_t st_blocks;	/* Number of 512-byte blocks allocated.  */
@@ -95,11 +95,11 @@
      * st_ctime but we don't have it under Linux.
      */
     __time_t st_atime;		/* Time of last access.  */
-    long int __reserved0;
+    unsigned long int st_atimensec;     /* Nscecs of last access.  */
     __time_t st_mtime;		/* Time of last modification.  */
-    long int __reserved1;
+    unsigned long int st_mtimensec;     /* Nsecs of last modification.  */
     __time_t st_ctime;		/* Time of last status change.  */
-    long int __reserved2;
+    unsigned long int st_ctimensec;     /* Nsecs of last status change.  */
     __blksize_t st_blksize;	/* Optimal block size for I/O.  */
     long int st_pad3;
     __blkcnt64_t st_blocks;	/* Number of 512-byte blocks allocated.  */
